ntbs-service/wwwroot/source/Components/ValidateMultiple.ts

```diff
@@ -0,0 +1,56 @@
+import Vue from 'vue';
+import { getHeaders, getValidationPath } from '../helpers';
+const axios = require('axios');
+
+const ValidateMultiple = Vue.extend({
+    props: {
+        model: String,
+        properties: Array
+    },
+    data: function() {
+        return {
+            hasError: false
+        }},
+    methods: {
+        // Annotate values to compare with ref="input[i]", and error fields with ref="errorField[i]", for any number of inputs that should be validated together
+        // Each error message will be mapped to the corresponding field.
+        validate: function () {
+            var inputs = this.createArrayFromRefElements('input');
+            var inputValues;
+            if (inputs[0].type == "radio") {
+                // TODO: Do this mapping for other types if element is reused for non-radio inputs.
+                inputValues = inputs.map((i: any) => i.checked);
+            }
+            var queryString = buildKeyValuePairsQueryString(this.properties, inputValues);
+            axios.get(`${getValidationPath(this.$props.model)}Properties?${queryString}`, null, { headers: getHeaders() })
+            .then((response: any) => {
+                var errorMessages = response.data;
+                this.hasError = errorMessages != '';
+                var errorFields = this.createArrayFromRefElements('errorField');
+                errorFields.forEach((errorField: any, index: number) => errorField.textContent = errorMessages[index]);
+            })
+            .catch((error: any) => {
+                console.log(error.response)
+            });
+        },
+        createArrayFromRefElements: function(elementName: string) {
+            var array = [];
+            for (var i = 0; i < this.properties.length; i++) {
+                array.push(this.$refs[`${elementName}[${i}]`]);
+            }
+            return array;
+        }
+    }
+});
+
+function buildKeyValuePairsQueryString(keys: Array<string>, values: Array<string>) {
+    var keyValueArray = [];
+    for (var i = 0; i < keys.length; i++) {
+        keyValueArray.push(`keyValuePairs[${i}][key]=${keys[i]}&keyValuePairs[${i}][value]=${values[i]}`)

```

Comment:
This is the notation required for the HTTP request to be correctly mapped to the serverside method :)

Reply:
Can we achieve this by passing a config object to `axios`?
https://github.com/axios/axios#request-config
Not sure if it'll use the same convention for lists, but worth trying if you havn't already, so we don't have to build our own strings!